server/src/test/java/org/elasticsearch/index/query/CommonTermsQueryBuilderTests.java

```diff
@@ -39,6 +40,13 @@
 
 public class CommonTermsQueryBuilderTests extends AbstractQueryTestCase<CommonTermsQueryBuilder> {
 
+    @After
+    public void validateDeprecationMessage() {
+        if (getTestName().equals("testParseFailsWithMultipleFields") == false) {
```

Comment:
Why is this needed ? Isn't it possible to check the warning in the `testParseFailsWithMultipleFields` test directly ?

Reply:
With the current approach, every call to the constructor logs a deprecation warning, so all the methods of this class including the ones inherited from the abstract superclass log this warning, apart from this `testParseFailsWithMultipleFields` which throws an exception in advance because of misconfiguration of the query.

Reply:
ok got it, thanks